swarm/network/stream/stream.go

```diff
@@ -388,14 +388,6 @@ func (r *Registry) Quit(peerId enode.ID, s Stream) error {
 	return peer.Send(context.TODO(), msg)
 }
 
-func (r *Registry) NodeInfo() interface{} {
```

Comment:
I figured this might be used by web3 JS interface, but I am not sure how defining there to return `nil` are useful.

Reply:
@zelig could you elaborate on why these cannot be deleted? I don't understand their usage. we have `admin_nodeInfo`, but I don't understand how this is related.

Reply:
@nonsense check https://github.com/ethersphere/go-ethereum/blob/e9b43afe6bf715774200fc0ae664c7c3658622be/p2p/server.go#L1053-L1055

every sub-protocol MAY (bot does NOT HAVE TO) implement additional `NodeInfo` for displaying in their own namespace. However, if it doesn't do anything (like in this case) it seems it can safely be skipped @zelig

Reply:
@holisticode exactly my point - why define them in the first place. These fields are `nil` by default, so why define methods that return `nil` - it makes no sense.

This is not an interface where you have to adhere to the signatures in the interface.

Additionally the code you pasted above shows that we are handling the case where these fields (func pointers) are not defined.

Reply:
@holisticode not to mention that we don't propagate `nodeInfo` or `peerInfo` as far as I can tell:

```
	if b.streamerSpec != nil && b.streamerRun != nil {
		protocol = append(protocol, p2p.Protocol{
			Name:    b.streamerSpec.Name,
			Version: b.streamerSpec.Version,
			Length:  b.streamerSpec.Length(),
			Run:     b.RunProtocol(b.streamerSpec, b.streamerRun),
		})
	}
```

Reply:
👍 remove @nonsense 
@holisticode you quoted it yourself: `if query := proto.NodeInfo; query != nil`
if it is removed - the condition will not be true and the sun will shine again